src/entity.c

```diff
@@ -412,12 +414,12 @@ ecs_type_t copy_from_prefab(
         }
 
         ecs_table_column_t *src_column = &prefab_columns[p + 1];
-        uint32_t size = src_column->size;
+        uint64_t size = src_column->size;
 
         if (size) {
             void *src_column_data = ecs_vector_first(src_column->data);
             void *src_ptr = ECS_OFFSET(
-                src_column_data, size * (prefab_index - 1));
+                src_column_data, size * ((uint64_t)prefab_index - 1));
```

Comment:
After updating the size variable, hopefully the need for this cast will go away.

Reply:
The issue is that ECS_OFFSET casts to uintptr_t which is 8 bytes, so it gives this warning:


Warning	C26451	Arithmetic overflow: Using operator '*' on a 4 byte value and then casting the result to a 8 byte value. Cast the value to the wider type before calling operator '*' to avoid overflow (io.2).